### PR TITLE
fix: preserve navigation functionality when `unload` event is deprecated

### LINKED ISSUE #29241: Understand/fix any implications of `unload` event deprecation within Chrome
### What would you like?

https://developer.chrome.com/docs/web-platform/deprecating-unload

Chrome is deprecating the upload event where it will begin ceasing to functions for some websites starting Q3 2024.

We should enable the flags to turn it off to understand the implications of the deprecation. https://github.com/fergald/docs/blob/master/explainers/permissions-policy-deprecate-unload.md#testing-with-chrome

We'll also need to handle users that have this flag manually turned on past the deprecation. 

I opened a PR to deprecate the unload event to see what would fail here: https://github.com/cypress-io/cypress/pull/29242

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by cacieprins**
- Closes https://github.com/cypress-io/cypress/issues/29241

### Additional details
In order to prepare for Chrome's removal of the `unload` event, we are migrating to using `pagehide` as a proxy for internal navigation detection. In order to preserve `window:unload`, for now, this event is also triggered on `pagehide`. Because the event emitted on `pagehide` is a superset of `Event`, which is emitted on `unload`, this is not a breaking change.

Future steps to support deprecation of `unload`:
* Add `window:pageshow` and `window:pagehide` events
* Warn when `window:unload` is used

### Steps to test
Existing tests should pass with the flags removed. To test full `unload` removal, uncomment the indicated flags in `packages/server/lib/util/chromium_flags.ts`.

### How has the user experience changed?
`window:unload` may fire both more reliably and at a slightly different point in the page lifecycle timeline. Otherwise, it should not be impacted.

### PR Tasks

- [X] Have tests been added/updated?
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
@cacieprins I'm not sure I would even advocate for exposing `page:show` and `page:hide` until users directly ask for it.

Shouldn't we update the `onUpload` callback to `onPageHide`?

**Review by jennifer-shehane — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

File: packages/app/src/runner/event-manager.ts [hunk 1]
```diff
@@ -344,7 +344,9 @@ export class EventManager {
     // when we actually unload then
     // nuke all of the cookies again
     // so we clear out unload
-    $window.on('unload', () => {
+    const unloadEvent = this.isBrowser({ family: 'chromium' }) ? 'pagehide' : 'unload'
```
Comment (jennifer-shehane) [inline packages/app/src/runner/event-manager.ts:350]: @cacieprins Can you add a note about why we decided to only target chromium here?

Comment (cacieprins) [reply packages/app/src/runner/event-manager.ts:350]: Absolutely

Comment (cacieprins) [reply packages/app/src/runner/event-manager.ts:350]: Added in https://github.com/cypress-io/cypress/pull/29525/commits/a64d262b417823c67ee405c6a4ddba052f81649d

**Pull request merged**